### PR TITLE
add info on TRL docs

### PR DESCRIPTION
**Pull request by lvwerra**
# What does this PR do?

As discussed offline, this adds a link to the `SFTTrainer` in `trl` in the `Trainer` section of the docs.

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Thanks a lot ! 🚀

**Review by ArthurZucker — APPROVED** [review]
LGTM

File: docs/source/en/main_classes/trainer.md [hunk 1]
```diff
@@ -18,6 +18,12 @@ rendered properly in your Markdown viewer.
 
 The [`Trainer`] class provides an API for feature-complete training in PyTorch for most standard use cases. It's used in most of the [example scripts](https://github.com/huggingface/transformers/tree/main/examples).
 
+<Tip>
+
+If you're looking to fine-tune a language model like Llama-2 or Mistral on a text dataset using autoregressive techniques, consider using [`trl`](https://github.com/huggingface/trl)'s [`~trl.SFTTrainer`]. This trainer is specially optimized for this particular task and supports sequence packing, LoRA, quantization, and DeepSpeed for efficient scaling to any model size. On the other hand, the [`Trainer`] is a more versatile option, suitable for a broader spectrum of tasks.
```
Comment (LysandreJik) [inline docs/source/en/main_classes/trainer.md:23]: Maybe mention it's a simple wrapper over `Trainer` but dedicated to SFT?

Comment (lvwerra) [reply docs/source/en/main_classes/trainer.md]: Tweaked the text a little bit. What do you think?

**Review by LysandreJik — APPROVED** [review]
Looks great!

**Comment by LysandreJik** [comment]
Feel free to merge once green!

**Pull request merged**